### PR TITLE
Addon (pod) deployment improvements.

### PR DESCRIPTION
**Pull request by jortel**
Resolves: https://github.com/konveyor/tackle2-hub/issues/184

In addition:
- Replaces  GET `/applications/1/identities` sub-resource with query parameters: `/identities?application=1&kind=source`.

### CODE REVIEW

**Review by mansam — APPROVED** [review]
LGTM

File: addon/task.go [hunk 1]
```diff
@@ -12,50 +12,66 @@ import (
...
+func (h *Task) Load() {
+	var err error
+	defer func() {
+		if err != nil {
+			panic(err)
+		}
+	}()
+	h.task = &api.Task{}
+	path := Params{api.ID: Settings.Addon.Task}.inject(api.TaskRoot)
+	err = h.client.Get(path, &h.task)
```
[the remark this answers is not in the document]
Comment (jortel) [reply addon/task.go]: No.  surprised it worked.
Fixed.

File: task/manager.go [hunk 2]
```diff
@@ -700,3 +685,24 @@ func (r *RuleIsolated) hasPolicy(task *model.Task, name string) (matched bool) {
...
 }
+
+//
+// Validator token validator.
+type Validator struct {
+	DB *gorm.DB
+}
+
+//
+// Valid token if task exits.
```
[the remark this answers is not in the document]
Comment (jortel) [reply task/manager.go]: Fixed.

File: auth/builtin.go [hunk 3]
```diff
@@ -33,7 +33,7 @@ func (r *NoAuth) Authenticate(token string) (jwToken *jwt.Token, err error) {
 // For the NoAuth provider, this just returns a single instance
 // of the NoAuthScope.
```
Comment (mansam) [inline auth/builtin.go:34]: Prob worth updating the comment to match the new behavior.

**Review by mansam — APPROVED** [review]

**Review by aufi — APPROVED** [review]

**Pull request merged**